tls/s2n_psk.h

```diff
@@ -89,9 +95,12 @@ int s2n_psk_calculate_binder(struct s2n_psk *psk, const struct s2n_blob *binder_
 int s2n_psk_verify_binder(struct s2n_connection *conn, struct s2n_psk *psk,
         const struct s2n_blob *partial_client_hello, struct s2n_blob *binder_to_verify);
 
-typedef int (*s2n_psk_selection_callback)(struct s2n_connection *conn, 
-                                          struct s2n_psk_identity *identities, size_t identities_length,
-                                          uint16_t *chosen_wire_index);                     
-/* This function will be labeled S2N_API and become a publicly visible api once we release the psk API. */
-int s2n_config_set_psk_selection_callback(struct s2n_connection *conn, s2n_psk_selection_callback cb);
+/* These functions and types will be publicly visible once we release the psk feature. */
 
+struct s2n_psk_identity_list;
+typedef int (*s2n_psk_selection_callback)(struct s2n_connection *conn,
+                                          struct s2n_psk_identity_list *psk_identity_list,
+                                          uint16_t *chosen_wire_index);
+int s2n_config_set_psk_selection_callback(struct s2n_config *config, s2n_psk_selection_callback cb);
+int s2n_psk_identity_list_next(struct s2n_psk_identity_list *list, uint8_t **next_identity, uint16_t *identity_size);
```

Comment:
I was wondering if we could create a struct `s2n_psk_identity` that wraps an s2n_blob in the similar way as we do for struct `s2n_psk_identity_list` over `s2n_stuffer` to send and receive `uint8_t **next_identity, uint16_t *identity_size` over here?

Reply:
Is the idea that we might want the customer to be able to retrieve values other than the identity for each element in the list? Like the PSK type?

Reply:
Yes, I believe it is a good use case for our customer, they may be interested in more than just the identity for each element in the list.